coderadar-test/src/test/java/io/reflectoring/coderadar/rest/useradministration/permissions/RemoveUserFromProjectControllerIntegrationTest.java

```diff
@@ -0,0 +1,56 @@
+package io.reflectoring.coderadar.rest.useradministration.permissions;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+
+import io.reflectoring.coderadar.graph.projectadministration.domain.ProjectEntity;
+import io.reflectoring.coderadar.graph.projectadministration.project.repository.ProjectRepository;
+import io.reflectoring.coderadar.graph.useradministration.domain.UserEntity;
+import io.reflectoring.coderadar.graph.useradministration.repository.UserRepository;
+import io.reflectoring.coderadar.rest.ControllerTestTemplate;
+import io.reflectoring.coderadar.useradministration.service.security.PasswordUtil;
+import java.util.Collections;
+import java.util.Date;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+class RemoveUserFromProjectControllerIntegrationTest extends ControllerTestTemplate {
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private ProjectRepository projectRepository;
+
+  private ProjectEntity testProject;
+  private UserEntity testUser;
+
+  @BeforeEach
+  void setUp() {
+    testProject = new ProjectEntity();
+    testProject.setVcsUrl("https://valid.url");
+    testProject.setName("project");
+    testProject.setVcsStart(new Date());
+    testProject.setVcsOnline(true);
+    testProject.setVcsPassword("testPassword");
+    testProject.setVcsUsername("testUser");
+    projectRepository.save(testProject);
+
+    testUser = new UserEntity();
+    testUser.setUsername("username");
+    testUser.setPassword(PasswordUtil.hash("password1"));
+    testUser.setProjects(Collections.singletonList(testProject));
+    userRepository.save(testUser);
+  }
+
+  @Test
+  void removeUserFromProjectSuccessfully() throws Exception {
+    mvc()
+        .perform(delete("/api/projects/" + testProject.getId() + "/users/" + testUser.getId()))
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andDo(document("user/role/project/remove"))
+        .andReturn(); // TODO: Document
```

Comment:
please handle this TODO

Reply:
TODO deleted as it was outdated. The endpoint is documented in line 51.